File: Miscellaneous/Python/66. Plus One.py
class Solution:
    def plusOne(self, digits):
        ans, carry = [], 1

        for i in range(len(digits)-1, -1, -1):
            sum = digits[i] + carry
            carry = sum // 10
            ans.append(sum % 10)
            sum  = 0

        if carry > 0:
            ans.append(carry)

        ans.reverse()

        return ans

obj = Solution()
digits = [1,2,3]
ans = obj.plusOne(digits)
print(ans)

File: Miscellaneous/Golang/66. Plus One.go
package main

import "fmt"

func main() {

	arr := []int{1, 2, 3}
	result := plusOne(arr)
	fmt.Println(result)
}

func plusOne(num []int) []int {
	var temp, ans []int
	carry := 1

	for i := len(num) - 1; i >= 0; i-- {
		sum := num[i] + carry
		carry = sum / 10
		temp = append(temp, sum%10)
	}
	if carry > 0 {
		temp = append(temp, carry)
	}

	idx := len(temp) - 1
	for i := 0; i < len(temp); i++ {
		if temp[i] == 0 {
			idx--
		}
	}

	for i := idx; i >= 0; i-- {
		ans = append(ans, temp[i])
	}

	return ans
}
